src/utilities.ts

```diff
@@ -147,7 +147,7 @@ export function subheaderOrigin(buffer: IOBuffer): SubheaderOrigin {
   const typeAndFlag = buffer.readUint32();
   /*not sure how to analyze flag yet */
   const flag = typeAndFlag.toString(2)[0] as '1' | '0';
-  const type = getListType(parseInt(typeAndFlag.toString(2).slice(17), 2));
+  const type = getListType(parseInt(typeAndFlag.toString(2).slice(1), 2));
```

Comment:
Those lines are a little bit strange. In javascript you have the bitwise operations: https://www.w3schools.com/js/js_bitwise.asp that are super fast.

If I'm not wrong it could be replaced by:
```
const flag = typeAndFlag >> 31;
const type = typeAndFlag & 0b01111111111111111111111111111111; // or ( 2 ** 31 - 1);
```